File: LeetCodeCPlusPlus/py53.cpp
// 最大子序和
// 和大于0保留，小于0重新开始。
#include <vector>
using namespace std;
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max = -2147483648,total=0;
        for (int i=0;i<nums.size();i++) {
            total += nums[i];
            max = total>max? total: max;
            if (total<0) total = 0;
        }
        return max;
    }
};

File: LeetCodePython/py53.py
# 最大子序和
# 动态规划；大于0继续加，小于0重新开始
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        result = -2147483648
        total = 0
        for num in nums:
            total += num
            result = total if total>result else result
            total = 0 if total<0 else total
        return result